integration_tests/src/main/python/udf_test.py

```diff
@@ -425,3 +427,42 @@ def test_func(spark):
             lambda data: [pd.DataFrame([len(list(data))])], schema="ret:integer")
 
     assert_gpu_and_cpu_are_equal_collect(test_func, conf=arrow_udf_conf)
+
+
+@pytest.mark.skipif(is_before_spark_350(), reason='mapInPandas with barrier mode is introduced by Pyspark 3.5.0')
+def test_map_in_pandas_with_barrier_mode():
+    def func(iterator):
+        tc = TaskContext.get()
+        assert tc is not None
+        assert not isinstance(tc, BarrierTaskContext)
+        for batch in iterator:
+            yield batch
+    with_gpu_session(lambda spark: spark.range(0, 1, 1, 1).mapInPandas(func, "id long", False).collect())
```

Comment:
Better to compare the collected results for CPU and GPU as we usually do.

Reply:
Done